layouts/joomla/edit/frontediting_modules.php

```diff
@@ -25,52 +25,50 @@
 $redirectUri  = '&return=' . urlencode(base64_encode(Uri::getInstance()->toString()));
 $target       = '_blank';
 $itemid       = Factory::getApplication()->input->get('Itemid', '0', 'int');
+$editUrl      = Uri::base() . 'administrator/index.php?option=com_modules&task=module.edit&id=' . (int) $mod->id;
 
-if (preg_match('/<(?:div|span|nav|ul|ol|h\d) [^>]*class="[^"]* jmoddiv"/', $moduleHtml))
-{
-	// Module has already module edit button:
-	return;
-}
-
-// Add css class jmoddiv and data attributes for module-editing URL and for the tooltip:
-$editUrl = Uri::base() . 'administrator/index.php?option=com_modules&task=module.edit&id=' . (int) $mod->id;
-
+// If Module editing site
 if ($parameters->get('redirect_edit', 'site') === 'site')
 {
 	$editUrl = Uri::base() . 'index.php?option=com_config&view=modules&id=' . (int) $mod->id . '&Itemid=' . $itemid . $redirectUri;
 	$target  = '_self';
 }
 
-// Add class, editing URL and tooltip, and if module of type menu, also the tooltip for editing the menu item:
+// Add link for editing the module
 $count = 0;
 $moduleHtml = preg_replace(
-	// Replace first tag of module with a class
-	'/^(\s*<(?:div|span|nav|ul|ol|h\d|section|aside|nav|address|article) [^>]*class="[^"]*)"/',
-	// By itself, adding class jmoddiv and data attributes for the URL and tooltip:
-	'\\1 jmoddiv" data-jmodediturl="' . $editUrl . '" data-target="' . $target . '" data-jmodtip="'
-	.	HTMLHelper::_('tooltipText',
-			Text::_('JLIB_HTML_EDIT_MODULE'),
-			htmlspecialchars($mod->title, ENT_COMPAT, 'UTF-8') . '<br>' . sprintf(Text::_('JLIB_HTML_EDIT_MODULE_IN_POSITION'), htmlspecialchars($position, ENT_COMPAT, 'UTF-8')),
-			0
-		)
-	. '"'
-	// And if menu editing is enabled and allowed and it's a menu module, add data attributes for menu editing:
-	.	($menusEditing && $mod->module === 'mod_menu' ?
-			' data-jmenuedittip="' . HTMLHelper::_('tooltipText', 'JLIB_HTML_EDIT_MENU_ITEM', 'JLIB_HTML_EDIT_MENU_ITEM_ID') . '"'
-			:
-			''
-		),
+	// Find first tag of module
+	'/^(\s*<(?:div|span|nav|ul|ol|h\d|section|aside|address|article) [^>]*>)/',
+	// Create and add the edit link and tooltip
+	'\\1 <a class="btn btn-link jmodedit" href="' . $editUrl . '" target="' . $target . '" aria-describedby="tip-' . (int) $mod->id . '">
+	<span class="icon-edit" aria-hidden="true"></span>' . Text::_('JGLOBAL_EDIT') . '</a>
+	<div role="tooltip" id="tip-' . (int) $mod->id . '">' . Text::_('JLIB_HTML_EDIT_MODULE') . '<br>' . htmlspecialchars($mod->title, ENT_COMPAT, 'UTF-8') . '<br>'. sprintf(Text::_('JLIB_HTML_EDIT_MODULE_IN_POSITION'), htmlspecialchars($position, ENT_COMPAT, 'UTF-8')) . '</div>',
 	$moduleHtml,
 	1,
 	$count
 );
 
-if ($count)
+// If menu editing is enabled and allowed and it's a menu module add link for editing
+if ($menusEditing && $mod->module === 'mod_menu')
 {
-	// Load stylesheet and javascript to head:
-	HTMLHelper::_('bootstrap.popover');
+// find the menu item id
```

Comment:
```suggestion
	// find the menu item id
```
Just a small, final code style change, it will not change RTC status if applied.